monkeytype/typing.py

```diff
@@ -251,8 +251,19 @@ def rewrite(self, typ):
         return typ
 
 
+class RewriteGenerator(TypeRewriter):
+    """Returns an Iterator, if the send_type and return_type of a Generator is None"""
+
+    def visit_Generator(self, typ):
+        args = getattr(typ, '__args__')
+        for e in args:
```

Comment:
Hmm, this doesn't look quite right to me. I would expect the generator `typ` to have `__args__` like `[Something, None, None]`, which would imply  that we would just want the next two lines, operating on `args` instead of `e.__args__`. Currently we're going two layers deep and I'm not sure why.